packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -134,6 +134,127 @@ def test_metrics_agents_statsd(dcos_api_session):
+    assert check_func(response) is True
+
+
+def test_task_metrics_metadata(dcos_api_session):
+    # install marathon on marathon
+    install_response = dcos_api_session.cosmos.install_package('marathon')
+    data = install_response.json()
+    dcos_api_session.marathon.wait_for_deployments_complete()
+
+    wait = True
+    mesos_id = node = ''
+    while wait:
+        master = dcos_api_session.masters[0]
+        state_response = dcos_api_session.get('/state', host=master, port=5050)
+        assert state_response.status_code == 200
+        state = state_response.json()
+        for framework in state['frameworks']:
+            if framework['name'] == 'marathon':
+                for task in framework['tasks']:
+                    if task['name'] == 'marathon-user':
+                        mesos_id = task['slave_id']
+                        wait = False
+                        break
+                break
+
+    for agent in state['slaves']:
+        if agent['id'] == mesos_id:
+            node = agent['hostname']
+            break
+
+    def check_task_metrics(response):
+        for line in response.text.splitlines():
+            if '#' in line:
+                continue
+            if 'marathon-user' in line:
+                return ('service_name="marathon"' in line and
+                        'task_name="marathon-user"' in line)
+        return False
+    check_metrics_prom(dcos_api_session, node, check_task_metrics)
+
+    # uninstall and cleanup framework
+    dcos_api_session.cosmos.uninstall_package('marathon', app_id=data['appId'])
+
+
+def test_executor_metrics_metadata(dcos_api_session):
+    # get count of all running containers to check against later for complete teardown
+    num_containers = 0
+    for agent_ip in dcos_api_session.slaves:
+        response = dcos_api_session.metrics.get('/containers', node=agent_ip)
+        assert response.status_code == 200
+        num_containers += len(response.json())
+
+    try:
+        # install hello-world framework
+        install_response = dcos_api_session.cosmos.install_package('hello-world', package_version='2.2.0-0.42.2')
+        data = install_response.json()
+
+        dcos_api_session.marathon.wait_for_deployments_complete()
+
+        wait = True
+        mesos_id = ''
+        while wait:
+            master = dcos_api_session.masters[0]
+            state_response = dcos_api_session.get('/state', host=master, port=5050)
+            assert state_response.status_code == 200
+            state = state_response.json()
+            for framework in state['frameworks']:
+                if framework['name'] == 'marathon':
+                    for task in framework['tasks']:
+                        if task['name'] == 'hello-world':
+                            mesos_id = task['slave_id']
+                            wait = False
+                            break
+                    break
```

Comment:
This stanza that blocks until the package is deployed could be factored out into a separate function. It's separable from the test and is duplicated in `test_task_metrics_metadata()`.

This would work nicely as a context manager. It could install a package and block until it's deployed on entry, then uninstall it and block until all its containers are finished on exit. Quick sketch:

```python
import contextlib


@contextlib.contextmanager
def dcos_package(dcos_api_session, package_name, package_version)
    app_id = dcos_api_session.cosmos.install_package(package_name, package_version=package_version).json()['appId']
    dcos_api_session.marathon.wait_for_deployments_complete()
    wait_for_package_deployment(dcos_api_session, package_name)
    yield
    dcos_api_session.cosmos.uninstall_package(package_name, app_id=app_id)
    wait_for_package_teardown(dcos_api_session, package_name, ...)  # Not sure what else this might need


def test_executor_metrics_metadata(dcos_api_session):
    with dcos_package(dcos_api_session, 'hello-world', '2.2.0-0.42.2'):
        executor_node = get_task_hostname(framework_name='marathon', task_name='hello-world')
        metrics_response = get_metrics_prom(dcos_api_session, executor_node)
        for line in metrics_response.text.splitlines():
            # assert stuff
    # done!
```